### PR TITLE
[PS-221] Update Mobile SDK with new Search Results URL API and new request ID parameter

### PR DESCRIPTION
**Pull request by swVisenze**

### CODE REVIEW

File: productcat/src/main/java/com/visenze/productcat/android/api/impl/SearchOperationsImpl.java [hunk 1]
```diff
@@ -60,6 +61,27 @@ public void imageSearch(ImageSearchParams params, final ProductCat.ResultListene
         imageSearch(params, resultListener, PRODUCT_SUMMARY_SEARCH);
     }
 
+    @Override
+    public void imageSearchResultPage(ImageSearchParams params, final ProductCat.ResultListener resultListener) {
+        byte[] imageBytes = null;
```
Comment (thehung111) [inline productcat/src/main/java/com/visenze/productcat/android/api/impl/SearchOperationsImpl.java]: the image validation is exactly the same as imageSearch method.
Can you put in a common validation method? Thanks.

Comment (thehung111) [reply productcat/src/main/java/com/visenze/productcat/android/api/impl/SearchOperationsImpl.java]: the only difference is the endpoint, should refactor for 2 methods to call the same base method with new parameter for endpoint

File: productcat/src/test/java/com/visenze/productcat/android/util/ResponseParserTest.java [hunk 2]
```diff
@@ -17,6 +17,23 @@
 
 public class ResponseParserTest {
 
+
+    @Test
+    public void parseSRPResult() {
+        String responseString = "{\"im_id\":\"202002063654d637a048cbb5106e7c28d05dacf41d9720acf38.jpg\",\"reqid\":\"05O1IQO3OSUHB8VCQ0Q536PG\",\"status\":\"OK\",\"error\":[],\"product_types\":[{\"type\":\"bag\",\"score\":0.862,\"box\":[32,2,185,218],\"attributes\":{}}],\"srp_url\":\"https://shopping.visenze.com/search/05O1IQO3OSUHB8VCQ0Q536PG?country=SG&amp;uid=4444\",\"recognize_result\":[{\"tag_group\":\"category\",\"tags\":[{\"tag_id\":\"root|bag\",\"tag\":\"bag\",\"score\":0.9836792349815369},{\"tag_id\":\"root|bag|backpack\",\"tag\":\"backpack\",\"score\":0.9464564323425293}]}]}";
+        ResultList resultList = ResponseParser.parseResult(responseString);
+
+        assertEquals("https://shopping.visenze.com/search/05O1IQO3OSUHB8VCQ0Q536PG?country=SG&amp;uid=4444", resultList.getSrpUrl());
```
Comment (thehung111) [inline productcat/src/test/java/com/visenze/productcat/android/util/ResponseParserTest.java]: can change this to "https://shopping.visenze.com/search/05O1IQO3OSUHB8VCQ0Q536PG?country=SG&uid=4444" and also the responseString.

The Wiki page auto convert "&" to "&amp;".

File: productcat/src/main/java/com/visenze/productcat/android/api/impl/SearchOperationsImpl.java (hunk 1 above)
Comment (swVisenze) [reply productcat/src/main/java/com/visenze/productcat/android/api/impl/SearchOperationsImpl.java]: changed already.

File: productcat/src/test/java/com/visenze/productcat/android/util/ResponseParserTest.java (hunk 2 above)
Comment (swVisenze) [reply productcat/src/test/java/com/visenze/productcat/android/util/ResponseParserTest.java]: changed already

**Pull request merged**